plugin/src/main/java/com/exadel/aem/toolkit/plugin/utils/StringUtil.java

```diff
@@ -1,182 +1,200 @@
+     * @param value      The array to convert
+     * @param valueType  {@code Class<?>} reference representing the underlying collection type
+     * @param conversion {@code Function<Object, String>} to convert every passed collection member before storing
+     * @param <T>        Type of the value
+     * @return String representing the value. If {@code null} reference was passed, empty string is returned
+     */
+    private static <T> String format(Collection<T> value, Class<?> valueType, Function<Object, String> conversion) {
+        if (value == null || value.isEmpty()) {
+            return StringUtils.EMPTY;
+        }
+        String collectionPart = value.stream()
+            .filter(Objects::nonNull)
+            .map(conversion)
+            .map(s -> s.startsWith(ARRAY_OPENING) && s.endsWith(ARRAY_CLOSING) ? ESCAPE_STRING + s : s)
+            .collect(Collectors.joining(CoreConstants.SEPARATOR_COMMA));
+        if (String.class.equals(valueType)) {
+            return String.format(ARRAY_TEMPLATE, collectionPart);
+        } else {
+            return String.format(TYPE_TOKEN_TEMPLATE, valueType.getSimpleName()) + String.format(ARRAY_TEMPLATE, collectionPart);
+        }
+    }
+
+    /* -------------------
+       Parsing collections
+       ------------------- */
+
+    /**
+     * Gets whether the passed string contains a JCR-compliant array/collection of values
+     * @param value The string to analyze
+     * @return True or false
+     */
+    public static boolean isCollection(String value) {
+        return ATTRIBUTE_LIST_PATTERN.matcher(value).matches();
+    }
+
+    /**
+     * Parses the provided string as a JCR-compliant array/collection and returns a {@code Set} of stored values
+     * @param value The string to analyze
+     * @return {@code Set} instance. If {@code null} was passed, an empty set is returned
+     */
+    public static Set<String> parseSet(String value) {
+        if (StringUtils.isEmpty(value)) {
+            return new HashSet<>();
+        }
+        return new LinkedHashSet<>(Arrays.asList(StringUtils
+            .strip(value, ARRAY_OPENING + ARRAY_CLOSING)
+            .split(SPLITTING_PATTERN)));
+    }
+
+    /**
+     * Parses the provided string as a JCR-compliant array/collection and returns a {@code List} of stored values
+     * @param value The string to analyze
+     * @return {@code List} instance. If {@code null} was passed, an empty list is returned
+     */
+    public static List<String> parseCollection(String value) {
+        if (StringUtils.isEmpty(value)) {
+            return new ArrayList<>();
+        }
+        return Arrays.asList(StringUtils
+            .strip(value, ARRAY_OPENING + ARRAY_CLOSING)
+            .split(SPLITTING_PATTERN));
+    }
+
+    /**
+     * Escape characters given a string.
+     * @param value The string to process
+     * @return escapedValue. The string with scaped values
+     * */
+    public static String escapeValue(String value) {

```

Comment:
Your approach is right!

Problem is , the name/placement of this method. 

You see: however this is a true utility method, it is not universal. A very ad-hoc method which escapes not "any" string but the very particular three symbols. And escapes them in a very particular situation - in `DependsOn` handler. 

Why I say "particular"? That;s because in an occasional other atrtributes of the same XML file we could have some `[..; .. ;..]`  structure and we wouldn't need it to be escaped in any manner because it wouldn't be handled by DependsOn => treated like a JS expression.

When I see such method in a common utility class, I may think this method is generic. Something like `encodeURI` method is JS. But it is not generic - it is for the particular case.

Therefore I'd recommend to move it to the very DependsOnHandler class and make `private static`.

Reply:
This was changed according with suggested!